packages/bitcore-lib-btg/lib/block/blockheader.js

```diff
@@ -219,16 +219,24 @@ BlockHeader.prototype.toBufferWriter = function toBufferWriter(bw) {
   if (!bw) {
     bw = new BufferWriter();
   }
-  bw.writeUInt32LE(this.version);
+  bw.writeInt32LE(this.version);
   bw.write(this.prevHash);
   bw.write(this.merkleRoot);
-  bw.writeUInt32LE(this.height);
-  bw.write(this.reserved);
+  // forkHeight 491407
+  // Hash calculation is different between prefork blocks and postfork blocks
+  if (this.height >= 491407) {
```

Comment:
We need to put the constant block number in the network configuration. Otherwise it will break testnet. (Testnet fork height = 0)